### ISSUE TITLE
COM registration fixing tool fails to register  X64 DLL's on ARM 64, and does not register native ARM 64 DLL's

### ISSUE THREAD

**Issue by mzanm** [nvaccess/nvda#16200]
### Steps to reproduce:
- Run the COM registration fixing tool from the tools menu of NVDA on an arm64-based system.

### Actual behavior:

a success message will appear, however an error sound plays and it appears in the log.

### Expected behavior:

No errors.

### NVDA logs, crash dumps and other attachments:

ERROR - RPC process 11496 (nvda_slave.exe) (21:21:37.063) - Dummy-4 (11808):
COMRegistrationFixes.register64bitServer:
Error registering C:\Windows\System32\oleaut32.dll in a 64-bit context: Command '['C:\\Windows\\Sysnative\\regsvr32.exe', '/s', 'C:\\Windows\\System32\\oleaut32.dll']' returned non-zero exit status 5.
ERROR - RPC process 11496 (nvda_slave.exe) (21:21:37.096) - Dummy-4 (11808):
COMRegistrationFixes.register64bitServer:
Error registering C:\Windows\System32\actxprxy.dll in a 64-bit context: Command '['C:\\Windows\\Sysnative\\regsvr32.exe', '/s', 'C:\\Windows\\System32\\actxprxy.dll']' returned non-zero exit status 5.

The issue appears to be that on windows on ARM 64, system32 points to native ARM64 DLL's, or it might point to ARM 32 since a special folder is used, but definitely Not X64 DLL's. Also, the COm registration fixing tool simply checks if the installation of Windows is 32 or 64 bit, it doesn't differentiate from X64 and ARM64.

#### NVDA installed/portable/running from source:
Installed.

#### NVDA version:
2024.1 beta 9
#### Windows version:
Windows 11 23H2 (ARM64) build 22631.3155

#### Name and version of other software in use when reproducing the issue:
None exactly, but i'm having issues with the native ARM versions of Microsoft Edge and VSCode where NVDA sometimes just says unknown and I'm hoping that running it can fix that.

#### Other information about your system:
Windows 11 virtualized on Parallels on Apple Silicon Macbook, I also have another installation on VMWare fusion and I have the same issue there.

#### Does the issue still occur after restarting your computer?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2023.3.3 has the same issue.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.

**Comment by LeonarddeR** [comment]
I think it makes sense to take this into account during the x64 migration, since the logic in the COM registration fixing tool will change. Therefore I"m adding the appropriate label.

**Comment by josephsl** [comment]
Hi,

January 2026 check-in: is anyone working on this?

Thanks.

**Comment by SaschaCowley** [comment]
@josephsl no-one at NV Access is currently working on this. @seanbudd and @michaelDCurran are the only ones with ARM64 machines, and they haven't had the time to look into it yet. If you have the hardware, please feel free to investigate

**Comment by seanbudd** [comment]
I'm currently working on this

**Comment by seanbudd** [comment]
From what I could tell, I could not reproduce the error when registering 64bit dlls with 2026.1.

Running the COM reg fixing tool with the beta branch yielded the following log:

```
COMRegistrationFixes.register64bitServer:
Registered C:\WINDOWS\System32\oleaut32.dll in a 64-bit context.
DEBUG - RPC process 34408 (nvda_slave.exe) (17:05:46.467) - Dummy-35 (17944):
COMRegistrationFixes.register64bitServer:
Registered C:\WINDOWS\System32\actxprxy.dll in a 64-bit context.
IO - inputCore.InputManager.executeGesture (17:05:47.650) - winInputHook (36056):
Input: kb(laptop):escape
```

Can anyone still reproduce this issue on [alpha/beta snapshots](https://download.nvaccess.org/snapshots/)?

**Comment by seanbudd** [comment]
Closing as fixed by the 64bit upgrade

**Comment by valiant8086** [comment]
Indeed, no error here on galaxy book go with today's alpha.
